src/cargo/core/compiler/future_incompat.rs

```diff
@@ -228,7 +228,7 @@ impl OnDiskReports {
     /// Returns an ANSI-styled report
     pub fn get_report(&self, id: u32, package: Option<&str>) -> CargoResult<String> {
         let report = self.reports.iter().find(|r| r.id == id).ok_or_else(|| {
-            let available = iter_join(self.reports.iter().map(|r| r.id.to_string()), ", ");
+            let available = itertools::join(self.reports.iter().map(|r| r.id), ", ");
```

Comment:
Is there a reason you preferred the free method over the extension trait method?

Reply:
The interface is similar to `iter_join` and can be benefit from `IntoIterator`.
(not applicable to this case though).